### PR TITLE
feat: update manifests with new prom image and delete flag

### PR DESCRIPTION
**Pull request by bwplotka**
Wanted clean state here and what I can import into latest manifest release.

### CODE REVIEW

**Review by TheSpiritXIII — APPROVED** [review]
Looks good. Just see my comment about the extra labels being added by helm!

File: manifests/operator.yaml [hunk 1]
```diff
@@ -18,6 +18,13 @@ apiVersion: scheduling.k8s.io/v1
 kind: PriorityClass
 metadata:
   name: gmp-critical
+  labels:
+    app.kubernetes.io/name: prometheus-engine
+    app.kubernetes.io/managed-by: Helm
```
Comment (pintohutch) [inline manifests/operator.yaml:23]: nit: I'd love to get rid of the "managed-by: Helm" label...

File: manifests/operator.yaml [hunk 2]
```diff
@@ -18,6 +18,13 @@ apiVersion: scheduling.k8s.io/v1
 kind: PriorityClass
 metadata:
   name: gmp-critical
+  labels:
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply manifests/operator.yaml:21]: will do, but first to main perhaps

[the remark this answers is not in the document]
Comment (TheSpiritXIII) [reply manifests/operator.yaml:21]: Looks like Adam got it: https://github.com/GoogleCloudPlatform/prometheus-engine/pull/948

**Comment by TheSpiritXIII** [comment]
Small nit: can we create a proper release/0.12 branch and push into that instead of release-please? I see release-please rebasing the branch and updating this PR. I'm not sure if it would keep changes such as these. Thanks!

**Review by TheSpiritXIII — APPROVED** [review]

**Comment by bwplotka** [comment]
Had to change Prometheus image to staging (chicken & egg issue), so I can test and reference new Prometheus image.

**Comment by bwplotka** [comment]
Updated, using SHA for now.

**Review by TheSpiritXIII — APPROVED** [review]

**Pull request merged**